libraries/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -451,6 +441,20 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     }
   }
 
+  private _parseDepPath(pkgPath: string): string {
+    let depPath: string = pkgPath;
+    if (this.shrinkwrapFileMajorVersion >= 6) {
+      const index: number = pkgPath.indexOf('@', pkgPath.indexOf('/@') + 2);
+      const suffixIndex: number | undefined = pkgPath.includes('(')
+        ? dp.indexOfPeersSuffix(pkgPath)
+        : undefined;
+      depPath = `${pkgPath.slice(0, index)}/${pkgPath.slice(index + 1, suffixIndex)}`;
+    }
+
+    const pkgInfo: ReturnType<typeof dp.parse> = dp.parse(depPath);
```

Comment:
Our coding convention is to spell out words instead of using arbitrary abbreviations, something like this:

- `import * as dp from '@pnpm/dependency-path';` -> `import * as dependencyPath`
- `_parseDepPath` -> `_parseDependencyPath`
- `pkgPath` -> `packagePath`

The underlying idea is that **Reading &gt; Writing**.  Being able to write source code easily is less important than other people being able to read it easily.